mailpoet/lib/EmailEditor/Engine/Renderer/BlocksRegistry.php

```diff
@@ -9,6 +9,12 @@ class BlocksRegistry {
 
   public function addBlockRenderer(string $blockName, BlockRenderer $renderer): void {
     $this->blockRenderersMap[$blockName] = $renderer;
+    add_filter('render_block_' . $blockName, [$renderer, 'render'], 10, 2);
+  }
```

Comment:
Just an idea. What if instead of registering ` [$renderer, 'render']` directly, we could register 
```php
 add_filter('render_block_' . $blockName, [$this, 'renderBlock'], 10, 2);`
 ```
  and then in render block  we could do:
 ```php
 public function renderBlock($blockContent, $parsedBlock) {
   $renderer = $this->blockRendererMap[$parsedBlock['name']]->render($blockContent, $parsedBlock, $potencialyAdditionalParameter);
 }
 ```
 I think this way we could allow passing `$settingsController` as a parameter to the render method rather than adding it as a dependency in the render parent class (done in follow up commit).
 
 I think it is more convenient because when some third-party block renderer wants to prepare their own renderer and wants to use data from `$settingsController` we will not force them to figure out where to get the settings controller to pass it to the constructor of the renderer.
 What do you say?

Reply:
Thank you for pointing this out.
I will try to refactor this behavior for better extensibility, as you recommended.